web/src/user_profile.js

```diff
@@ -306,18 +306,13 @@ export function register_click_handlers() {
 
         function removal_success(data) {
             if (data.removed.length > 0) {
-                // Most of the work for handling the unsubscribe is done
-                // by the subscription -> remove event we will get.
-                // However, the user profile component has not yet
-                // implemented live update, so we do update its
-                // UI manually here by removing the stream from this list.
-                $stream_row.remove();
-
                 ui_report.success(
                     $t_html({defaultMessage: "Unsubscribed successfully!"}),
                     $alert_box,
                     1200,
                 );
+
+                update_user_profile_streams_list(target_user_id);
```

Comment:
Hmm, I would have thought we'd have this call be in the `server_events` code path, rather than here; see the "new application feature" tutorial for the multi-browser reasons why we generally don't want a click handler to do UI rendering updates like this.

Reply:
So I reviewed the code again and I realized we actually don't want this line over here since we are already handling the case of updating the stream_list when any server event happens so in case the user is unsubscribed then the `server_events_dispatch.js` file will update the stream_list so at last we don't want to do any change here and hence removed it.

Reply:
Great, thanks!